dbms/src/Storages/System/StorageSystemTables.cpp

```diff
@@ -30,6 +31,9 @@ static ColumnPtr getFilteredDatabases(const ASTPtr & query, const Context & cont
     for (const auto & db : context.getDatabases())
         column->insert(db.first);
 
+    std::string temporary_database = "";
+    column->insert(temporary_database);
+
     Block block { ColumnWithTypeAndName( std::move(column), std::make_shared<DataTypeString>(), "database" ) };
```

Comment:
You could also put column `temporary` into the block.
It allows excluding the temporary database in case of `WHERE temporary != 0`